src/cid.js

```diff
@@ -229,24 +230,28 @@ export class CID {
    * @returns {CID<Data, Format, Alg, Version>|null}
    */
   static asCID (input) {
+    if (input == null) {
+      return null
+    }
+
     const value = /** @type {any} */ (input)
     if (value instanceof CID) {
       // If value is instance of CID then we're all set.
       return value
-    } else if (value != null && value.asCID === value) {
-      // If value isn't instance of this CID class but `this.asCID === this` is
-      // true it is CID instance coming from a different implementation (diff
-      // version or duplicate). In that case we rebase it to this `CID`
-      // implementation so caller is guaranteed to get instance with expected
-      // API.
+    } else if ((value['/'] != null && value['/'] === value.bytes) || value.asCID === value) {
+      // If value isn't instance of this CID class but `this.asCID === this` or
+      // `value['/'] === value.bytes` is true it is CID instance coming from a
+      // different implementation (diff version or duplicate). In that case we
+      // rebase it to this `CID` implementation so caller is guaranteed to get
+      // instance with expected API.
       const { version, code, multihash, bytes } = value
       return new CID(
         version,
         code,
         /** @type {API.MultihashDigest<Alg>} */ (multihash),
-        bytes || encodeCID(version, code, multihash.bytes)
+        value['/'] ?? bytes ?? encodeCID(version, code, multihash.bytes)
```

Comment:
I'm not sure I understand the intention here & mutating passed in `value` makes me really uncomfortable. Unless there is good reason could you please leave original line:

```suggestion
        bytes || encodeCID(version, code, multihash.bytes)
```

Or if there is a good reason to mutate, please add a comment explaining intention

Reply:
There's no mutation here.  `??` only uses the `right` value if `left` is `null` or `undefined`, `||` does the same but on falsy values so also `false`, `''` or `0` so generally `??` is safer.

That said, there's not much difference, I have reverted it.